### PR TITLE
Add several quantized models for pose estimation and action recognition

### PR DESCRIPTION
**Pull request by wuxun-zhang**
This PR firstly aims to apply calibration tool for pose estimation and action recognition models. And then we provide several pretrained INT8 models in modelzoo. Overall we can get 2.5x~5.9x speedup with fusion/VNNI optimization and less than 0.5% of accuracy loss.

@pengzhao-intel @xinyu-intel @zhreshold @bryanyzhu

### CODE REVIEW

**Comment by mli** [comment]
Job PR-1000-2 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1000/2/index.html 
Code coverage of this PR:  vs. Master:

File: scripts/action-recognition/test_recognizer.py [hunk 1]
```diff
@@ -157,7 +190,7 @@ def test(ctx, val_data, opt, net):
             pred = net(X.astype(opt.dtype, copy=False))
             if opt.use_softmax:
                 pred = F.softmax(pred, axis=1)
-            pred = F.mean(pred, axis=0, keepdims=True)
+            # pred = F.mean(pred, axis=0, keepdims=True)
```
[the remark this answers is not in the document]
Comment (bryanyzhu) [reply scripts/action-recognition/test_recognizer.py]: @wuxun-zhang I changed the dataloader for video action recognition a month ago, it might only support `batch_size=1` during testing at this moment, but I'm not sure. Let me look into this.

[the remark this answers is not in the document]
Comment (wuxun-zhang) [reply scripts/action-recognition/test_recognizer.py]: I just commented out this, and it works well when batch-size>1.

**Comment by mli** [comment]
Job PR-1000-3 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1000/3/index.html 
Code coverage of this PR:  vs. Master:

File: scripts/action-recognition/test_recognizer.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bryanyzhu) [reply scripts/action-recognition/test_recognizer.py]: The reason it works when `batch-size>1` if you comment out that line, is because the default `num_segments=1` and default data augmentation during test is `single center crop`. But if we have more segments or using ten-crop, it will fail too. 

For example, if you use default setting, the label is 64x1 vector, and the prediction is also 64x1 vector, so it doesn't raise any error. However, if you use three segments and ten-crop, then the label is still 64x1, but the prediction now is 1920x1. The shape does not match. 

The test code and dataloader right now only support `batch_size=1`. Since this is not high priority, I didn't look into this for a long time. If you need `batch_size>1`, I can go ahead fix it and do a PR. This will take a day or two.

[the remark this answers is not in the document]
Comment (wuxun-zhang) [reply scripts/action-recognition/test_recognizer.py]: I see. Thanks for your explanation. 

Just running test mode on CPU, `batch_size>1` will bring some speedup over `batch_size=1`. If it expects not too much effort, please help fix this to support `batch_size>1`. Of course, it is not urgent since it is not the dependency of this PR. I will add this line back.

**Comment by mli** [comment]
Job PR-1000-4 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1000/4/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1000-5 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1000/5/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1000-7 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1000/7/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1000-8 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1000/8/index.html 
Code coverage of this PR:  vs. Master:

**Comment by zhreshold** [comment]
Could you solve the conflicts?

**Comment by wuxun-zhang** [comment]
@zhreshold Yes, will do this. However, there is another problem. This PR has hard dependency on [PR#16695](https://github.com/apache/incubator-mxnet/pull/16695) , but looks that current mxnet nightly build is experiencing failure. I cannot pin mxnet on the latest version. So perhaps we can run rebase later untill mxnet nightly build is ok.

**Comment by mli** [comment]
Job PR-1000-9 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1000/9/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1000-10 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1000/10/index.html 
Code coverage of this PR:  vs. Master:

**Comment by wuxun-zhang** [comment]
@bryanyzhu Just rebased code against master. Please help review/merge. Thanks.

**Comment by bryanyzhu** [comment]
@wuxun-zhang Will do the review ASAP. Thank you for the PR.

File: scripts/action-recognition/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+### FP32 inference
+
+```
+
+export CPUs=`lscpu | grep 'Core(s) per socket' | awk '{print $4}'`
+export OMP_NUM_THREADS=${CPUs}
+export KMP_AFFINITY=granularity=fine,noduplicates,compact,1,0
+
+# dummy data
+python test_recognizer.py --model inceptionv3_ucf101 --use-pretrained --use-tsn --mode hybrid --input-size 299 --new-height 340 --new-width 450 --num-segments 3 --batch-size 64 --benchmark
```
Comment (bryanyzhu) [inline scripts/action-recognition/README.md]: `--use-tsn` is deprecated, please remove it. Right now, if you set `--num-segments` larger than 1, TSN mode will be automatically turned on.

File: scripts/action-recognition/README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+# real data
+python test_recognizer.py --model inceptionv3_ucf101 --use-pretrained --use-tsn --mode hybrid --input-size 299 --new-height 340 --new-width 450 --num-segments 3 --batch-size 64
```
Comment (bryanyzhu) [inline scripts/action-recognition/README.md]: Similarly, remove `use-tsn`.

File: scripts/action-recognition/README.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+```
+
+### Calibration
+
+Naive calibrated model by using 5 batch data (32 images per batch). Quantized model will be saved into `./model/`.
+
+```
+# ucf101 dataset
+python test_recognizer.py --model inceptionv3_ucf101 --new-height 340 --new-width 450 --input-size 299 --num-segments 3 --use-pretrained --use-tsn --calibration
```
Comment (bryanyzhu) [inline scripts/action-recognition/README.md]: Similarly, remove `--use-tsn`.

File: scripts/action-recognition/README.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+export KMP_AFFINITY=granularity=fine,noduplicates,compact,1,0
+
+# dummy data
+python test_recognizer.py --model inceptionv3_ucf101 --mode hybrid --input-size 299 --new-height 340 --new-width 450 --batch-size 64 --num-segments 3 --quantized --benchmark
+
+# real data
+python test_recognizer.py --model inceptionv3_ucf101 --mode hybrid --input-size 299 --new-height 340 --new-width 450 --batch-size 64 --num-segments 3 --quantized
+
+# deploy static model
+python test_recognizer.py --model ucf --deploy --model-prefix ./model/inceptionv3_ucf101-quantized-naive --input-size 299 --new-height 340 --new-width 450 --batch-size 64 --num-segments 3 --benchmark
```
Comment (bryanyzhu) [inline scripts/action-recognition/README.md]: As far as I can recall, we don't have a model named `ucf`. Can you explain why you use `--model ucf`? Thank you.

File: scripts/action-recognition/README.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+Users are also recommended to bind processes to specific cores via `numactl` for better performance, like below:
+
+```
+numactl --physcpubind=0-27 --membind=0 python test_recognizer.py ...
+```
+
+## Performance
+Below results are collected based on Intel(R) VNNI enabled C5.12xlarge with 24 physical cores.
+
+|model | fp32 Top-1 | s8 Top-1 |
```
Comment (bryanyzhu) [inline scripts/action-recognition/README.md]: Why s8? I think it is int8. Correct me if s8 has another meaning. Thank you.

File: scripts/action-recognition/README.md (hunk 5 above)
Comment (wuxun-zhang) [reply scripts/action-recognition/README.md]: In `deploy` mode, we only need to specify `--model-prefix` for loading saved model, at this time whatever the value of `--model` is fine. However, maybe we need give a more meaningful name here, like `inceptionv3_ucf101`.

File: scripts/action-recognition/README.md (hunk 6 above)
Comment (wuxun-zhang) [reply scripts/action-recognition/README.md]: Yes, it is int8. I'll fix it. Thanks

File: scripts/action-recognition/README.md (hunk 5 above)
Comment (bryanyzhu) [reply scripts/action-recognition/README.md]: I see. You can delete `--model` or use `inceptionv3_ucf101 ` to be clearer. I'm fine with either way.

File: scripts/action-recognition/README.md (hunk 2 above)
Comment (wuxun-zhang) [reply scripts/action-recognition/README.md]: Done

**Comment by mli** [comment]
Job PR-1000-11 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1000/11/index.html 
Code coverage of this PR:  vs. Master:
